asdf/block.py

```diff
@@ -823,16 +824,27 @@ def array_storage(self):
         return self._array_storage
 
     @property
-    def compression(self):
-        return self._compression
+    def input_compression(self):
+        """
+        The compression codec used to read the block.
+        """
+        return self._input_compression
 
-    @compression.setter
-    def compression(self, compression):
-        self._compression = mcompression.validate(compression)
+    @input_compression.setter
+    def input_compression(self, compression):
+        self._input_compression = mcompression.validate(compression)
 
     @property
-    def is_compressed(self):
-        return self._compression is not None
+    def output_compression(self):
+        """
+        The compression codec used to write the block.
+        :return:
+        """
+        return self._output_compression
```

Comment:
The thing is, if we change the code this way, setting `all_array_compression=None` will not work - `self._output_compression` will be None and `self._input_compression` will take over. I actually looked into this and decided that it's better to be explicit.

If you think that having the output compression default to the input compression, we could add "input" compression type, which can be the default and mean exactly that.